proto/client-api/src/lib.rs

```diff
@@ -1,13 +1,57 @@
 //! Client API protocol implementation for Medea media server.
 
-// TODO: when using enum's Self will be in stable, remove it.
-#![allow(clippy::use_self)]
-
 use std::collections::HashMap;
 
+use macro_attr::*;
 use medea_macro::dispatchable;
+use newtype_derive::*;
 use serde::{de::Deserializer, ser::Serializer, Deserialize, Serialize};
 
+macro_attr! {
+    /// ID of [`Peer`].
+    #[cfg_attr(
+        feature = "medea",
+        derive(Deserialize, Debug, Hash, Eq, Default, PartialEq)
+    )]
+    #[cfg_attr(feature = "jason", derive(Serialize))]
+    #[derive(Clone, Copy, NewtypeDisplay!)]
+    pub struct PeerId(pub u64);
+}
+
+macro_attr! {
+    /// ID of [`MediaTrack`].
+    #[cfg_attr(
+        feature = "medea",
+        derive(Deserialize, Debug, Hash, Eq, Default, PartialEq)
+    )]
+    #[cfg_attr(feature = "jason", derive(Serialize))]
+    #[derive(Clone, Copy, NewtypeDisplay!)]
```

Comment:
It seems that `#[derive(Display)]` from `derive_more` without any additional attrs will play much better here.